frappe/event_streaming/doctype/event_update_log/event_update_log.py

```diff
@@ -140,3 +140,134 @@ def check_docstatus(out, old, new, for_child):
+	:param consumer: The EventConsumer doc
+	"""
+	if update_log.update_type == 'Create':
+		# consumer is obviously up to date
+		return True
+
+	prev_logs = frappe.get_all(
+		'Event Update Log',
+		filters={
+			'ref_doctype': update_log.ref_doctype,
+			'docname': update_log.docname,
+			'creation': ['<', update_log.creation]
+		},
+		order_by='creation desc',
+		limit_page_length=1
+	)
+
+	if not len(prev_logs):
+		return False
+
+	prev_log_consumers = frappe.get_all(
+		'Event Consumer Selector',
+		fields=['consumer'],
+		filters={
+				'parent': prev_logs[0].name,
+				'parenttype': 'Event Update Log',
+				'consumer': consumer.name
+		}
+	)
+
+	return len(prev_log_consumers) > 0
+
+
+def mark_consumer_read(update_log_name, consumer_name):
+	"""
+	This function appends the Consumer to the list of Consumers that has 'read' an Update Log
+	"""
+	update_log = frappe.get_doc("Event Update Log", update_log_name)
+	if len([x for x in update_log.consumers if x.consumer == consumer_name]):
+		return
+
+	frappe.get_doc(frappe._dict(
+			doctype="Event Consumer Selector",
+			consumer=consumer_name,
+			parent=update_log_name,
+			parenttype="Event Update Log",
+			parentfield="consumers"
+	)).insert(ignore_permissions=True)
+
+
+def get_unread_update_logs(consumer_name, dt, dn):
+	"""
+	Get old logs unread by the consumer on a particular document
+	"""
+	already_consumed = [x[0] for x in frappe.db.sql("""
+		SELECT
+			update_log.name
+		FROM `tabEvent Update Log` update_log
+		JOIN `tabEvent Consumer Selector` consumer ON consumer.parent = update_log.name
+		WHERE
+			consumer.consumer = %(consumer)s
+	""", {"consumer": consumer_name}, as_dict=0)]
+
+	logs = frappe.get_all(
+			"Event Update Log",
+			fields=['update_type', 'ref_doctype',
+							'docname', 'data', 'name', 'creation'],
+			filters={
+					"ref_doctype": dt,
+					"docname": dn,
+					"name": ["not in", already_consumed]
+			},
+			order_by="creation"
+	)
+
+	return logs
+
+
+@frappe.whitelist()
+def get_update_logs_for_consumer(event_consumer, doctypes, last_update):
+	"""
+	Fetches all the UpdateLogs for the consumer
+	It will inject old un-consumed Update Logs if a doc was just found to be accessible to the Consumer
+	"""
+
+	if isinstance(doctypes, str):
+		doctypes = frappe.parse_json(doctypes)
+	
+	from frappe.event_streaming.doctype.event_consumer.event_consumer import has_consumer_access
+
+	consumer = frappe.get_doc("Event Consumer", event_consumer)
+	docs = frappe.get_list(
+			doctype='Event Update Log',
+			filters={'ref_doctype': ('in', doctypes),
+							 'creation': ('>', last_update)},
+			fields=['update_type', 'ref_doctype',
+							'docname', 'data', 'name', 'creation'],
+			order_by='creation desc'
+	)
+
+	result = []
+	to_update_history = []
+	for d in docs:
+		if (d.ref_doctype, d.docname) in to_update_history:
+			# will be notified by background jobs
+			continue
+
+		if not has_consumer_access(consumer=consumer, update_log=d):
```

Comment:
```suggestion
		if d.update_type != 'Delete' and not has_consumer_access(consumer=consumer, update_log=d):
```
We should skip the `has_consumer_access` check for delete type update because in delete type we only store the docname and not the data in order to evaluate the condition

Reply:
@ruchamahabal 
Thank you for the review!
Handling deletion along with event-conditions is a bit tricky like you mentioned. The approach i took for consumer access to a `Delete-Event Update Log` is to check if the same consumer had access to the last/previous  `Event Update Log` of the same document.

Please check this section of code ?
https://github.com/frappe/frappe/pull/10868/files/6f2cb82ba27ec8479c997edad3731e0527f85403#diff-cc533e59861e7bd38d177abdb6440516R160